internal/datacoord/segment_info.go

```diff
@@ -508,9 +508,10 @@ func (s *SegmentInfo) getSegmentSize() int64 {
 	return s.size.Load()
 }
 
-// getDeltaCount use cached value when segment is immutable
+// L1 segment deltaCount changes in any state
+// L0 segment deltaCount won't change
 func (s *SegmentInfo) getDeltaCount() int64 {
-	if s.deltaRowcount.Load() < 0 || s.State != commonpb.SegmentState_Flushed {
+	if s.deltaRowcount.Load() <= 0 || s.GetLevel() != datapb.SegmentLevel_L0 {
```

Comment:
Make make `getDeltaCount` calculate every call for L0 segment. Looks like not by design?

Reply:
this will calculate each time for No L0 segment

Reply:
For flushed segment, it shall be calculated only once.

Reply:
But flushed segments delete rowcounts changes all the time. Besides this functions will be called by L0 segments only.